### DISCUSSION TITLE
Any way to remove the Ctrl C terminal command?

### DISCUSSION THREAD

**Discussion by 9wow** [oobabooga/text-generation-webui#5793] [Q&A]
Is there any way to prevent the webui CMD from terminating the process with Ctrl + C? I just lost all my settings because I was trying to copy a seed from the terminal.

**Comment by TiagoTiago** [comment]
I imagine maybe somewhere around where the code prints out "Received Ctrl+C. Shutting down Text generation web UI gracefully.", a modification could be done where it will provide a prompt in the terminal requiring the user to confirm they wanna exit instead of just going ahead and "shutting down gracefully".


Btw, if you still had the web UI open in the browser, there's a chance you could just write down/screenshot the settings and re-apply them next time; a few times I've even been able to simply relaunch it and just hit the save/apply/whatever button to save stuff I didn't had time to save before an accidental Ctrl-c or an unrelated crash.

**Comment by Woisek** [comment]
> Is there any way to prevent the webui CMD from terminating the process with Ctrl + C? I just lost all my settings because I was trying to copy a seed from the terminal.

But you know, that you don't do copy (& paste) with CTRL + C in the terminal, right ... ?

**Reply by TiagoTiago** [reply]
For the sake of completeness, the correct shortcut in most cases, for terminals, is `Ctrl-Shift-c`, and likewise for pasting, you also add `Shift` to the combo as well (be mindful if you have multiple lines in the clipboard, after each line the terminal will react as if you had pressed Enter when pasting; so beware of what you have in the clipboard before pasting and where you are pasting it).

**Reply by Woisek** [reply]
The safest way to copy in the terminal would be to mark and right click.
But the main thing is, why copy from the terminal in the first place ... ?

**Reply by TheLounger** [reply]
> For the sake of completeness...

For Windows Terminal it's CTRL+C (only when something is selected) and CTRL+V, which is actually really nice.

**Reply by TiagoTiago** [reply]
> > For the sake of completeness...
> 
> For Windows Terminal it's CTRL+C (only when something is selected) and CTRL+V, which is actually really nice.


Oh, been a few years since I had a Windows machine; interesting they have changed this since back in the old days. From what I remember, they did had a "select text mode" from the menu which would let you mark text and would automatically copy it to the clipboard once you finished marking (or I think maybe you needed to press enter to make stick), and menu option to paste; plus the two modified keyboard controls. And I think for a long while it also inherited the DOS shortcuts of `Ctrl-Insert` for copy, `Shift-Insert`  for pasting, and `Ctrl-Del` for cutting.

**Reply by TheLounger** [reply]
Nothing changed, Windows Terminal is just new/different, compared to Powershell/Command prompt.

**Reply by 9wow** [reply]
> don't CTRL + C in the terminal, right ... ?

why not? It works fine when i need to copy something out of it.

**Reply by 9wow** [reply]
@Woisek 
> But the main thing is, why copy from the terminal in the first place ... ?

read the post...

**Comment by TheLounger** [comment]
CTRL+C is handled here.
https://github.com/oobabooga/text-generation-webui/blob/1a7c027386f43b84f3ca3b0ff04ca48d861c2d7a/server.py#L65C1-L70C45
Guess it could be an option to disable it, or you could just think twice before smashing buttons 😂

**Reply by 9wow** [reply]
> twice before smashing buttons

i wasn't aware that doing a shortcut that copies text = smashing buttons.
